File: packages/lib-model-js/src/test/TestableResource/TestableResource.models.ts
import { type EntityResourceModel } from '@lib/model/core/EntityResource/EntityResource.models';

export type TestableResourceModel = EntityResourceModel & {
  date?: Date;

  group: string;

  index: number;

  number?: number;

  string: string;

  stringArray?: Array<string>;

  stringOptional?: string;
};

File: packages/lib-model-py/src/lib_model/test/testable_resource/testable_resource_models.py
from __future__ import annotations

from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel, ConfigDict, Field, RootModel


class TestableResourceModel(BaseModel):
    model_config = ConfigDict(
        extra='allow',
    )
    date: Optional[datetime] = None
    group: str
    index: float
    number: Optional[float] = None
    string: str
    stringArray: Optional[List[str]] = None
    stringOptional: Optional[str] = None
    field_id: str = Field(..., alias='_id')
    created: datetime
    isFixture: Optional[bool] = None
    beforeCreate: None = None


class Model(RootModel[TestableResourceModel]):
    root: TestableResourceModel
